org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/JDTLanguageServer.java

```diff
@@ -475,6 +476,13 @@ public void didOpen(DidOpenTextDocumentParams params) {
 	@Override
 	public void didChange(DidChangeTextDocumentParams params) {
 		logInfo(">> document/didChange");
+		try {
+			Job.getJobManager().join(DocumentLifeCycleHandler.DOCUMENT_LIFE_CYCLE_JOBS, new NullProgressMonitor());
```

Comment:
While the fix works as expected, I'm a bit worried about joining on a job without possibilities to cancel. Are there any other alternative?